sympy/solvers/solveset.py

```diff
@@ -191,12 +216,18 @@ def _invert_real(f, g_ys, symbol):
                 return _invert_real(base, res, symbol)
 
         if not base_has_sym:
-            if base is not S.Zero:
+            rhs = g_ys.args[0]
+            if base is not S.Zero and rhs > 0:
                 return _invert_real(expo,
                     imageset(Lambda(n, log(n)/log(base)), g_ys), symbol)
-            elif g_ys.args[0] is S.One:
+            elif base is not S.Zero and rhs < 0:
+                powe = power(base, -rhs)
+                if powe is not False and powe%2 == 0:
```

Comment:
You can do `base**int(log(-rhs)/log(base)) == -rhs`, though I'm not sure which method will be faster. Also, there may be similar functions in the number theory module.

Reply:
> Also, there may be similar functions in the number theory module.

I looked in number theory and there is no such function. I think number theory can have such function and it can be imported from there?

Reply:
Your problem is of a much narrower scope: you are looking for `base**(2*p) == -rhs` which means that `-rhs` is a square number. So you are looking for `(base**p) == sqrt(-rhs)`. I suspect it makes sense to first affirm that `-rhs` is a perfect square -- this should be fast. Then you check for the multiplicity of `base` in `sqrt(-rhs)`:

```
s, b = integer_nthroot(-rhs, 2)
if b:
    m = multiplicity(base, s)
    if pow(base, m) == s:
        return expo, FiniteSet(2*m)
```

Reply:
This makes more sense and is quite fast. I will update the branch.